server/api/offline_batch_results.py

```diff
@@ -185,6 +185,89 @@ def record_offline_batch_results(
     return jsonify(status="ok")
 
 
+@api.route(
+    "/election/<election_id>/jurisdiction/<jurisdiction_id>/round/<round_id>/results/batch/",
+    methods=["POST"],
+)
+@restrict_access([UserType.JURISDICTION_ADMIN])
+def record_offline_batch_results_add(
+    election: Election,  # pylint: disable=unused-argument
+    jurisdiction: Jurisdiction,
+    round: Round,
+):
+    offline_batch_results = request.get_json()
+    validate_offline_batch_results(election, jurisdiction, round, offline_batch_results)
+
+    for batch_results in offline_batch_results:
+        for contest_choice_id, result in batch_results["choiceResults"].items():
+            db_session.add(
+                OfflineBatchResult(
+                    jurisdiction_id=jurisdiction.id,
+                    batch_name=batch_results["batchName"],
+                    batch_type=batch_results["batchType"],
+                    contest_choice_id=contest_choice_id,
+                    result=result,
+                )
+            )
+
+    db_session.commit()
+
+    return jsonify(status="ok")
+
+
+@api.route(
+    "/election/<election_id>/jurisdiction/<jurisdiction_id>/round/<round_id>/results/batch/<batch_name>",
```

Comment:
what if we just give each batch a uuid so editing batch names gets easier?

Reply:
maybe as a future improvement but... I *really* don't want to modify the data model right now.